src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransPatterns.java

```diff
@@ -501,7 +500,10 @@ private void handleSwitch(JCTree tree,
             MethodSymbol bsm = rs.resolveInternalMethod(tree.pos(), env, syms.switchBootstrapsType,
                     bootstrapName, staticArgTypes, List.nil());
 
-            Type resolvedSelectorType = seltype;
+            Type resolvedSelectorType = syms.objectType;
```

Comment:
isn't this change assigning Object to `resolvedSelectorType` for more cases than before JEP455? not sure if there is a semantic change here

Reply:
Correct. I also moved it at the top of this method.